### PR TITLE
[Demo] Update diagram

### PR DESCRIPTION
**Pull request by julianocosta89**
This PR adds the missing `flagd` service to the diagram and realigns all services.
Also, we have removed redis from the demo some time ago and replaced with ValKey.

### CODE REVIEW

**Comment by julianocosta89** [comment]
@open-telemetry/demo-approvers it took me a while to align all services in a way the lines weren't too overlapping, LMK what you think.

**Comment by julianocosta89** [comment]
Preview available here: https://deploy-preview-5678--opentelemetry.netlify.app/docs/demo/architecture/

File: content/en/docs/demo/architecture.md [hunk 1]
```diff
@@ -14,12 +14,13 @@ graph TD
 subgraph Service Diagram
 accountingservice(Accounting Service):::dotnet
 adservice(Ad Service):::java
-cache[(Cache<br/>&#40redis&#41)]
+cache[(Cache<br/>&#40ValKey&#41)]
```
Comment (mviitane) [inline content/en/docs/demo/architecture.md:17]:
```suggestion
cache[(Cache<br/>&#40Valkey&#41)]
```

**Review by mviitane — APPROVED** [review]
Good improvement!

**Review by svrnm — APPROVED** [review]

**Pull request merged**